### PR TITLE
docs(mempool/lanes): Update ADR with latest design changes

### LINKED ISSUE #4120: mempool/lanes: Update ADR after initial design
- [ ] ~~Extend ADR with alternative approaches to obtain lanes info from app.~~
- [x] Clarify in ADR that higher priorities have higher values.
- [x] Explain why we need to put limits to the capacities of each lane.
- [ ] ~~Add property "if tx1 is appended to lane1 and then tx2 is appended to lane2, with priority(lane1) > priority(lane2), then txs maintain the FIFO ordering when accessed with WRR".~~

### PR DESCRIPTION
**Pull request by hvanz**
Closes #4120 
- [Allow different lanes to the same priority](https://github.com/cometbft/cometbft/commit/73603656d5ef27223531f00b37b0fc4a9d9bf325)
- [Limits on lane capacities](https://github.com/cometbft/cometbft/commit/2de8f077dbb11ba9fdde9b676808f8d56b2fd1ed)

### CODE REVIEW

File: docs/references/architecture/adr-118-mempool-lanes.md [hunk 1]
```diff
@@ -462,24 +462,38 @@ configurations are:
...
+partitioning the mempool space.
+
+1. Proportionally to lane priorities: This approach could lead to underutilization of the mempool if
+   there are significant discrepancies between priority values, as it would allocate space unevenly.
+2. Evenly across all lanes: Assuming high-priority transactions are smaller in size than
+   low-priority transactions, this approach would still allow for more high-priority transactions to
+   fit in the mempool compared to lower-priority ones.
+
+For the MVP we have chosen the second approach. Note that the maximum lane capacities apply both to
+the number of transactions and the size in bytes.
```
Comment (jmalicevic) [inline docs/references/architecture/adr-118-mempool-lanes.md:477]: Should we mention here the option the user has to increase the mempool capacity if they see that the assigned slot is too small ? Sergio mentioned that during the meeting where we discussed this.

Comment (jmalicevic) [reply docs/references/architecture/adr-118-mempool-lanes.md:477]: Not sure whether this should go into the ADR or in a potential user guide but the idea that for a first attempt to mitigate the big transactions attack, users could have two lanes of the same priority but assign big txs only to one of those so they do not fill up the mempool.

File: docs/references/architecture/adr-118-mempool-lanes.md [hunk 2]
```diff
@@ -359,27 +359,27 @@ mempool. The following is a summary of the key design decisions:
...
-information returned by the application only contains priorities, which are used to identify lanes.
+The field `lane_priorities` is a map from lane identifiers to priorities. Different lanes may have
+the same priority. On the mempool side, lane identifiers will mainly be used for user interfacing
+(logging, metric labels).
 
-The lowest priority a lane may have is 1. The value 0 is reserved for two cases: when the
-application does not classify the transaction (i.e. no priority returned) and for `CheckTx`
-responses of invalid transactions.
+The lowest priority a lane may have is 1. Higher values correspond to higher priorities. The value 0
+is reserved for two cases: when the application does not classify the transaction (i.e. no priority
```
[the remark this answers is not in the document]
Comment (hvanz) [reply docs/references/architecture/adr-118-mempool-lanes.md]: I updated the text. I hope it's more clear now.

File: docs/references/architecture/adr-118-mempool-lanes.md (hunk 1 above)
Comment (hvanz) [reply docs/references/architecture/adr-118-mempool-lanes.md]: I added your first suggestion to the text. The second one I would leave it for the user guide.

File: docs/references/architecture/adr-118-mempool-lanes.md [hunk 3]
```diff
@@ -359,33 +360,34 @@ mempool. The following is a summary of the key design decisions:
...
+happens either when the application does not classify transactions, or when the transaction is
+invalid.
 
 On receiving the information from the app, CometBFT will validate that:
-- `lane_priorities` has no duplicates (values in `lane_priorities` don't need to be sorted),
-- `default_lane_priority` is in `lane_priorities` (the default lane is not necessarily the lane with
-  the lowest priority), and
-- the list `lane_priorities` is empty if and only if `default_lane_priority` is 0.
+- `default_lane` is a key in `lane_priorities`, and
+- `lane_priorities` is empty if and only if `default_lane` is 0.
```
Comment (hvanz) [inline docs/references/architecture/adr-118-mempool-lanes.md:390]: `default_lane` is empty, not 0

**Review by jmalicevic — APPROVED** [review]

**Pull request merged**